src/main/ruby/core/posix.rb

```diff
@@ -40,4 +40,30 @@ def self.attach_function(name, argument_types, return_type)
   attach_function :access, [:string, :int], :int
   attach_function :chmod, [:string, :mode_t], :int
   attach_function :chown, [:string, :uid_t, :gid_t], :int
+  attach_function :dup, [:int], :int
+  attach_function :dup2, [:int, :int], :int
+  attach_function :fchmod, [:int, :mode_t], :int
+  attach_function :fchown, [:int, :uid_t, :gid_t], :int
+  attach_function :fsync, [:int], :int
+
+  attach_function :getegid, [], :gid_t
+  attach_function :geteuid, [], :uid_t
+  attach_function :getgid, [], :gid_t
+  attach_function :getuid, [], :uid_t
+  attach_function :seteuid, [:uid_t], :int
+  attach_function :setgid, [:gid_t], :int
+  attach_function :setuid, [:uid_t], :int
+
+  if Rubinius.linux?
+    attach_function :__errno_location, [], :pointer
+    def self.errno_address; __errno_location; end
+  elsif Rubinius.darwin?
+    attach_function :__error, [], :pointer
+    def self.errno_address; __error; end
+  elsif Rubinius.solaris?
+    attach_function :___errno, [], :pointer
+    def self.errno_address; ___errno; end
+  else
+    raise 'Unsupported platform'
+  end
```

Comment:
We could read/write errno trivially in C if we make a small shared library, which we would then call with NFI, but I didn't want to introduce a shared library in this PR.